### ISSUE TITLE
Will `read.xlsx` support selecting  specific class type of variables?

### ISSUE THREAD

**Issue by eolembrain-leo** [awalker89/openxlsx#58]
It will be very helpful when this function provide parameter like colClasses="character", instead of guessing the class type of the variable, which might be not a robust way of reading data.

**Comment by skanskan** [comment]
I thought it was already implemented but the colClasses argument produces this error:
 unused argument (colClasses = ...)

**Comment by richierocks** [comment]
This feature would be very useful, particularly case where there are multiple data types within a column in the spreadsheet.  I really like the interface used by `data.table::fread`'s `colClasses` argument, where you can specify columns using a character vector or a list.

A related point is to be able to choose how large integers are read into R: either as `numeric`, `bit64::integer64` or character.  Again, `fread` has a sensible way of dealing with this.

**Comment by kasselbe** [comment]
An advantage of openxlsx is that it does not need Java, like packages xlsconnect or xlsx.
In my experience Java can consume huge amounts of memory for some tasks.
Compared to package readxl (Hadley Wickham) function read.xlsx in package openxlsx
offers more functionality, with one surprising exception: there is no function argument to
define the type of the data columns to read.
The other packages do offer such an argument: colTypes (in XLConnect), colClasses (in xlsx),
col_types (in readxl).
Do you plan to add a column types argument to function read.xlsx?
This would be very welcome!

**Comment by yOb1990** [comment]
Yes it's mandatory in some cases. For example, I have a column wich is a mix of characters and numbers, but open.xlsx considers the column as number. I have other troubles with some mixed date type columns. Other packages offer this option to specify the col types before opening, but for a huge excel I have java memory problems.
The only way right now is "openxlsx" ! Please if you can implement it ! :)

**Comment by leoluyi** [comment]
@yOb1990 maybe [`readxl`](https://readxl.tidyverse.org/) would be a better solution for reading Excel by now.

**Comment by awalker89** [comment]
@yOb1990 Can you share an example xlsx file where characters are being read an numbers?
